Source/RxSwift/unwrap.swift

```diff
@@ -16,8 +16,17 @@ extension ObservableType {
 
      - returns: An observable sequence of non-optional elements
      */
-
+    @available(*, deprecated, renamed: "compactMap", message: "compactMap was added to the main RxSwift repository and should be used instead of unwrap")
     public func unwrap<T>() -> Observable<T> where Element == T? {
-        return self.filter { $0 != nil }.map { $0! }
+        return self.compactMap({ $0 })
+    }
+
+    /**
+     Takes a sequence of optional elements and returns a sequence of non-optional elements, filtering out any nil values.
+
+     - returns: An observable sequence of non-optional elements
+     */
+    public func compactMap<T>() -> Observable<T> where Element == T? {
```

Comment:
We probably shouldn't add this. Just deprecate the original. Thanks :)

Reply:
Ok, I'll remove it then. 
Also I'll wait for the next release of RxSwift including compactMap on sharedSequence Single and Maybe In order to deprecate all the old code at once !
(https://github.com/ReactiveX/RxSwift/pull/1978)